src/components/AppProvider/AppProvider.tsx

```diff
@@ -85,23 +77,21 @@ export class AppProvider extends React.Component<AppProviderProps, State> {
 
     // eslint-disable-next-line react/no-did-update-set-state
     this.setState({
-      link: linkComponent,
-      intl: new I18n(i18n),
       appBridge: createAppBridge({shopOrigin, apiKey, forceRedirect}),
     });
   }
 
   render() {
-    const {theme = {logo: null}, children} = this.props;
-    const {intl, appBridge, link} = this.state;
+    const {i18n, linkComponent, theme = {logo: null}, children} = this.props;
+    const {appBridge} = this.state;
 
     return (
-      <I18nContext.Provider value={intl}>
+      <I18nContext.Provider value={new I18n(i18n)}>
```

Comment:
is it a good idea to recreate `I18n` on every render? I think it's better to memorize this value

Reply:
`i18n` is an array or object, so it's compared by reference, so it's not going to be easy to memo. We could serialize it, or use a deep compare however it may be more expensive than creating a new `I18n`. And considering `AppProvider` should be at the top level of an application, it's unlikely to see many renders